misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -0,0 +1,58 @@
+package misk.jobqueue
+
+import misk.tokens.TokenGenerator
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class FakeJobQueue @Inject constructor(
+  private val jobHandlers: Map<QueueName, JobHandler>,
+  private val tokenGenerator: TokenGenerator
+) : JobQueue {
+  private val jobQueues = ConcurrentHashMap<QueueName, ConcurrentLinkedDeque<FakeJob>>()
+
+  override fun enqueue(
+    queueName: QueueName,
+    body: String,
+    deliveryDelay: Duration?,
+    attributes: Map<String, String>
+  ) {
+    val id = tokenGenerator.generate()
+    val job = FakeJob(queueName, id, body, attributes)
+    jobQueues.getOrPut(queueName, ::ConcurrentLinkedDeque).add(job)
+  }
+
+  fun peekJobs(queueName: QueueName) : List<Job> {
+    val jobs = jobQueues[queueName]
+    return jobs?.toList() ?: listOf()
+  }
+
+  fun handleJobs(queueName: QueueName) {
+    val jobHandler = jobHandlers[queueName]!!
+    val jobs = jobQueues[queueName] ?: return
+
+    while (true) {
+      val job = jobs.poll() ?: break
```

Comment:
This should probably support a timeout or your tests might hang when there are bugs.

Reply:
@mightyguava added a timeout

Reply:
I actually meant you should pass the timeout into the `poll()`. That's the thing that might block indefinitely if you try to consume jobs that never got queued. Or I might be misunderstanding how this package is to be used.

Reply:
ahhh, I thought you might be worried about a self-enqueuing job or similar. `poll()` is non-blocking and returns null if the queue is empty.

Reply:
Oh, I totally misunderstood. That makes sense. Thanks

Reply:
I'm just going to get rid of this. As-is It's not that useful since the jobs are running in the same thread. If it is problematic we can start handling the fake jobs in a separate thread and interrupt it after a timeout.